docs/learn/web-development/es-modules.md

```diff
@@ -16,16 +16,49 @@ export function doBar() {
 }
 ```
 
-You can load them from an HTML page using a module script:
+In contrast to regular scripts, variables declared inside modules are scoped to the file itself. This makes them especially useful for writing modular code, explicitly declaring the API surface using exports.
+
+## Loading modules
+
+### Module scripts
+
+You can load javascript modules using scripts with `type="module"`. You can reference the module file using `src`, or write the module directly inline.
 
 ```html
 <html>
   <body>
     <script type="module" src="./app.js"></script>
+    <script type="module">
+      import './app.js';
+
+      console.log('hello world from my module');
+    </script>
   </body>
 </html>
 ```
 
+### Static imports
+
+A module can import other modules using static imports. These imports are executed eagerly, the browser will download the statically imported modules before running the code in your module.
+
+```js
+import { foo } from './foo.js';
+```
+
+### Dynamic imports
+
+You can also import other modules using dynamic `import()` function. This import is executed lazily, the browser will download the file only when the function is executed.
+
+```js
+function loadComponent() {
+  return import('./components/my-component.js');
+}
+```
+
+## File extensions
+
+You will often see modules using the `.mjs` file extension. For the browser, the file extension doesn't matter, but for node js `.mjs` is an indication to execute this code as a module. If the `package.json` has `type="module"` set, node js it will execute `.js` file as a module automatically as well.
```

Comment:
Is it worth discussing file extension support at the server level here?

Reply:
Yep, I added this in.